go/tendermint/tendermint.go

```diff
@@ -408,17 +412,45 @@ func (t *tendermintService) GetGenesis() *genesis.Document {
 	return t.genesis
 }
 
-func (t *tendermintService) ForceInitialize() error {
+func (t *tendermintService) InitializeBackends() (epochtimeAPI.Backend, error) {
 	t.Lock()
 	defer t.Unlock()
 
-	var err error
+	timeSource, err := epochtime.New(t)
+	if err != nil {
+		t.Logger.Error("InitializeBackends: failed to initialize epochtime backend",
+			"err", err,
+		)
+		return nil, err
+	}
+	t.timeSource = timeSource
+
+	// XXX: with https://github.com/oasislabs/oasis-core/issues/1879 initialize
+	// other backends here as well.
+
+	return timeSource, nil
+}
+
+func (t *tendermintService) ForceInitialize() error {
```

Comment:
If you cave the part that actually does the lazy init into a separate function, you can get rid of the kludges around locking.  Eg:
```
  doLazyInit := func() error {
    t.Lock()
    defer t.Unlock()

    // Do stuff.
  }
  if err := doLazyInit(); err != nil {
    return err
  }

  // Start consensus backends.
}
```

Reply:
:+1: